cmd/admin-console.go

```diff
@@ -47,173 +39,35 @@ var adminConsoleFlags = []cli.Flag{
-	}
-}
-
-// Extend madmin.LogInfo to add String() and JSON() methods
-type logMessage struct {
-	Status string `json:"status"`
-	madmin.LogInfo
-}
-
-// JSON - jsonify loginfo
-func (l logMessage) JSON() string {
-	l.Status = "success"
-	logJSON, err := json.MarshalIndent(&l, "", " ")
-	fatalIf(probe.NewError(err), "Unable to marshal into JSON.")
-
-	return string(logJSON)
-}
-
-func getLogTime(lt string) string {
-	tm, err := time.Parse(time.RFC3339Nano, lt)
-	if err != nil {
-		return lt
-	}
-	return tm.Format(logTimeFormat)
-}
-
-// String - return colorized loginfo as string.
-func (l logMessage) String() string {
-	var hostStr string
-	b := &strings.Builder{}
-	if l.NodeName != "" {
-		hostStr = fmt.Sprintf("%s ", colorizedNodeName(l.NodeName))
-	}
-	log := l.LogInfo
-	if log.ConsoleMsg != "" {
-		if strings.HasPrefix(log.ConsoleMsg, "\n") {
-			fmt.Fprintf(b, "%s\n", hostStr)
-			log.ConsoleMsg = strings.TrimPrefix(log.ConsoleMsg, "\n")
-		}
-		fmt.Fprintf(b, "%s %s", hostStr, log.ConsoleMsg)
-		return b.String()
-	}
-	if l.API != nil {
-		apiString := "API: " + l.API.Name + "("
-		if l.API.Args != nil && l.API.Args.Bucket != "" {
-			apiString = apiString + "bucket=" + l.API.Args.Bucket
-		}
-		if l.API.Args != nil && l.API.Args.Object != "" {
-			apiString = apiString + ", object=" + l.API.Args.Object
-		}
-		apiString += ")"
-		fmt.Fprintf(b, "\n%s %s", hostStr, console.Colorize("API", apiString))
-	}
-	if l.Time != "" {
-		fmt.Fprintf(b, "\n%s Time: %s", hostStr, getLogTime(l.Time))
-	}
-	if l.DeploymentID != "" {
-		fmt.Fprintf(b, "\n%s DeploymentID: %s", hostStr, l.DeploymentID)
-	}
-	if l.RequestID != "" {
-		fmt.Fprintf(b, "\n%s RequestID: %s", hostStr, l.RequestID)
-	}
-	if l.RemoteHost != "" {
-		fmt.Fprintf(b, "\n%s RemoteHost: %s", hostStr, l.RemoteHost)
-	}
-	if l.UserAgent != "" {
-		fmt.Fprintf(b, "\n%s UserAgent: %s", hostStr, l.UserAgent)
-	}
-	if l.Trace != nil {
-		if l.Trace.Message != "" {
-			fmt.Fprintf(b, "\n%s Error: %s", hostStr, console.Colorize("LogMessage", l.Trace.Message))
-		}
-		if l.Trace.Variables != nil {
-			for key, value := range l.Trace.Variables {
-				if value != "" {
-					fmt.Fprintf(b, "\n%s %s=%s", hostStr, key, value)
-				}
-			}
-		}
-		if l.Trace.Source != nil {
-			traceLength := len(l.Trace.Source)
-			for i, element := range l.Trace.Source {
-				fmt.Fprintf(b, "\n%s %8v: %s", hostStr, traceLength-i, element)
-			}
-		}
-	}
-	logMsg := strings.TrimPrefix(b.String(), "\n")
-	return fmt.Sprintf("%s\n", logMsg)
+	Name:               "console",
+	Usage:              "show MinIO logs",
+	Action:             mainAdminConsole,
+	OnUsageError:       onUsageError,
+	Before:             setGlobalsFromContext,
+	Flags:              append(adminConsoleFlags, globalFlags...),
+	Hidden:             true,
+	HideHelpCommand:    true,
+	CustomHelpTemplate: "This command is deprecated and will be removed in a future release. Use 'mc support logs show' instead.\n",
```

Comment:
As the custom help says here, `mc admin console` command is only deprecated, we should still support it with a warning message like this. I think this is the right behavior until the time comes to kill the command completely.

But the current behavior is in between not supported "killed" state and still alive with a warning message state.

So, if the user uses the old command with some mistake in defining the flags, we show full supported usage with  adminConsole and global flags (because code in lines 28~39 is still live).
```
$ mc admin console --typ
mc: <ERROR> Invalid command usage, flag provided but not defined: -typ

SUPPORTED FLAGS:
   --limit value, -l value       show last n log entries (default: 10)
   --type value, -t value        list error logs by type. Valid options are '[minio, application, all]' (default: "all")
   --config-dir value, -C value  path to configuration folder (default: "/home/ersan/.mc")
   --quiet, -q                   disable progress bar display
   --no-color                    disable color theme
   --json                        enable JSON lines formatted output
   --debug                       enable debug output
   --insecure                    disable SSL certificate verification
   --help, -h                    show help
```

If the user chooses to get `--help`, then we display this warning message:
```
$ mc admin console -h
This command is deprecated and will be removed in a future release. Use 'mc support logs show' instead.
```

If the user puts together the old command in correct syntax, the command fails and we recommend the new command to be used, that is basically means the old command is already dead.
```
$ mc admin console --type all myminio
mc: Please use 'mc support logs show --type all myminio'
```

We should adjust our behavior and choose either deprecation or dead/not supported state in all these little scenarios.

Reply:
Thank for the detailed analysis @ebozduman . The moto of this PR is to replace `mc admin console` with `mc support logs show`. We are stopping the use of old command and asking users to switch to new command. There is no deprecation. 

1. Corrected the help command for `mc admin console`
```
ashish@ashish:~/code/go/src/github/minio/mc$ ./mc admin console -h
This command is replace by 'support logs show' command. Please use 'mc support logs show'.
```
2. If some one type a wrong flag then they need to correct it first and then we show the new command. 

3. The limit flag is replace by last . so if some one writes `mc admin console --limit 5 myminio2` , the suggestion shows the correct command `lease use 'mc support logs show --last 5 myminio2`. 
 ```
ashish@ashish:~/code/go/src/github/minio/mc$ ./mc admin console --limit 5 myminio2
mc: Please use 'mc support logs show --last 5 myminio2'
```